ui/ts/components/table.ts

```diff
@@ -41,41 +58,101 @@ module Components {
        * The columns in the table will be displayed in the same order as the
        * returned array.
        */
-      columns: () => TableColumn<T>[];
+      columns: Utils.ReadOnlyProperty<TableColumn<T>[]>;
       /* rows is a function that returns an array of row data for display in the
        * table. 
-       *
-       * Rows will be displayed in the same order as the returned array.
        */
-      rows: () => T[];
+      rows: Utils.ReadOnlyProperty<T[]>;
     }
 
     class Controller<T> {
       data: TableData<T>;
+      sortedRows: Utils.ReadOnlyProperty<T[]>;
+      private _sortColumn: Utils.Property<TableColumn<T>> = Utils.Prop(null);
+      private _sortAscend: Utils.Property<boolean> = Utils.Prop(false);
 
       constructor(data: TableData<T>) {
         this.data = data;
+
+        this.sortedRows = Utils.Computed(data.rows, this._sortColumn, this._sortAscend, (rows: T[], sortCol: TableColumn<T>, asc: boolean): T[] => {
+          let result = _(rows);
+          if (sortCol && sortCol.sortable) {
+            // Sort the rows using the currently selected column, if the column
+            // is sortable. Sort by the output of the column's sortValue()
+            // method if specified, using the output of view() otherwise.
+            if (sortCol.sortValue) {
+              result = result.sortBy(sortCol.sortValue);
+            } else {
+              result = result.sortBy(sortCol.view);
+            }
+
+            if (asc) {
+              result = result.reverse();
```

Comment:
result is still a set of chained instructions, it doesn't convert into an array unless value() is called.